crates/sqlexec/src/metastore/catalog.rs

```diff
@@ -47,6 +48,98 @@ pub struct SessionCatalog {
     /// Map schema IDs to objects in the schema.
     schema_objects: HashMap<u32, SchemaObjects>,
 }
+pub struct AsyncSessionCatalog {
+    inner: Arc<RwLock<SessionCatalog>>,
+}
```

Comment:
The introduction of async to all these methods is going to be very difficult to resolve (and not needed for most this). The only things that need async is `maybe_refresh_state` and `mutate`, and that's just due to the client.

Some changes we can make here:
- Move `metastore_client` to this struct.
- Move implementation of `mutate` and `maybe_refresh_state` to this struct.
  - Scope locking down to the `set_state` and `version` calls.
- Replace tokio lock with parking_lot.

Once that's in, we can remove the asyncs on the functions that don't need them.